src/graph.impl.cc

```diff
@@ -605,6 +607,24 @@ namespace hpp {
         }
       }
 
+      void Graph::removeCollisionPairFromEdge
+      (ID edgeId, const char* joint1, const char* joint2) throw (hpp::Error)
+      {
+        graph::EdgePtr_t edge = getComp <graph::Edge> (edgeId);
+
+	try {
+	  using hpp::core::RelativeMotion;
+	  RelativeMotion::matrix_type& m (edge->relativeMotion ());
+	  DevicePtr_t robot = getRobotOrThrow (problemSolver());
+	  JointIndex i1 = robot->model ().getJointId (joint1);
+	  JointIndex i2 = robot->model ().getJointId (joint2);
```

Comment:
`getJointId` does not throw when the joint name does not exist. You have two options:
- check that `0 <= i1 < robot->model().joints.size()`
- use `robot->getJointByName(joint1)->jointId()` which checks that the joint exists. (Check in hpp::pinocchio::Joint for the correct method to get the joint ID, I do not remember it).